phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1214,6 +1216,8 @@ private void sendBatch(Map<TableRef, MultiRowMutationState> commitBatch, long[]
                     serverTimeStamps == null ?
                         validateAndGetServerTimestamp(tableRef, multiRowMutationState) :
                         serverTimeStamps[i++];
+                long syscatTime = EnvironmentEdgeManager.currentTimeMillis() - startTime;
```

Comment:
@mnpoonia This is not the only place where we call validateAndGetServerTimestamp. It is also called from here https://github.com/apache/phoenix/blob/master/phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java#L895

Reply:
Also, this is being called in a loop so you need to reset the value of startTime for every iteration.

Reply:
Agreed. instead of keeping the time here i pushed it down to the method validateAndGetServerTimestamp.

Reply:
@tkhurana Let me know  if it looks good now.